ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -506,10 +478,23 @@ protected Boolean findNextCompactionAndExecute(boolean computeStats) {
         task (currently we're using Tez split grouping).
         */
         QueryCompactor queryCompactor = QueryCompactorFactory.getQueryCompactor(t, conf, ci);
+        boolean generic_stats = conf.getBoolVar(HiveConf.ConfVars.HIVE_COMPACTOR_GATHER_STATS);
```

Comment:
Why did you move the config evaluation inside of the compaction loop? 
Please use camelCase naming convention.

Reply:
I moved it back